src/main/java/org/folio/circulation/support/Clients.java

```diff
@@ -247,5 +260,20 @@ private CollectionResourceClient createCalendarStorageClient(
     throws MalformedURLException {
     return getCollectionResourceClient(client, context, "/calendar/periods");
   }
-  
+
+  private CollectionResourceClient createPatronNoticePolicesStorageClient(
+    OkapiHttpClient client,
+    WebContext context)
+    throws MalformedURLException {
+    return getCollectionResourceClient(client, context,
+      "/patron-notice-policy-storage/patron-notice-policies");
```

Comment:
Probably not a question for this PR, but why are patron notice policies in mod-circulation-storage? It doesn't really make sense to me, especially with fees/fines data in the policy. This seems like something that should be in its own module. These microservices are starting to become monolithic in my mind. Maybe I am missing something. Sorry for the rant... 😞

Reply:
@mreno-EBSCO I don’t think you are missing something. 

I think the modules have grown, partly due to the overhead of introducing new modules and partly due to behaviour that can only be accomplished when interfaces are provided by the same module (e.g. constraints in storage).

This is something that likely needs discussing in a more general sense. It may be worth raising a JIRA issue to try to raise awareness of examples like this, and decide upon a policy, of when to move interface implementations to new modules. What do you think?

I think that mod-circulation knowing about the policies is itself a compromise. I’d prefer that mod-circulation was responsible for identifying and publishing the checked in (to return) and checked out (both deliberately past tense) events and a different module was responsible for determining if a policy meant that a notification should be sent.

However I don’t think we have the infrastructure in place for a design like that at the moment.

Reply:
@marcjohnson-kint I hear you. I've been ranting about this in the past, but there is just too much to do to get basic v1 functionality working that this type of compromise gets swept under the rug, which I totally get.

I think what adds some level of confusion is that, to me, it is not obvious that `patron-notice-policy-storage` would be in mod-circulation-storage. I had to do a folio.org GitHub search to find it. The raml file in mod-circulation-storage doesn't include "-storage" (probably an oversight) in the name, so that is confusing as well.

Anyway, I think we need a coordinated service that can manage internal out-of-band operations across all modules. How we achieve this is probably best left until after v1 or until necessity dictates. Much to my chagrin, this is probably not that case.

Reply:
> it is not obvious that patron-notice-policy-storage would be in mod-circulation-storage

Indeed, that is partially deliberate in that the interface name shouldn't reveal the implementor, but it does make finding them harder.

> The raml file in mod-circulation-storage doesn't include "-storage" (probably an oversight) in the name, so that is confusing as well.

That is likely a mistake, that could do with fixing, though it might impact our documentation generation scripts. And there are quite a lot of minor issues like this with RAML and JSON files across the repositories.

> I think we need a coordinated service that can manage internal out-of-band operations across all modules. How we achieve this is probably best left until after v1 or until necessity dictates.

By out of band do you mean some kind of publish / subscribe mechanism and/or something based upon messaging rather than HTTP?

If you do, then I believe there have been some discussions about pub / sub as I believe some work has started on a capability like that. I agree, exploring this topic more is definitely a post 2019 Q1 conversation.

Reply:
By out of band, I meant something not triggered by a REST API call, like on a schedule. Though, this service could be fed by a pub/sub queue, like Kafka. Or use reddis... or workflow tasks... or whatever. 😄 I've been part of the pub/sub discussions and have worked with the AES module. We've got a long way to go, but we are making some progress. Agreed, this is all post Q1 and possibly post v1.